### DISCUSSION TITLE
Sorting out Model Components

### DISCUSSION THREAD

**Discussion by HenryGeorgist** [HydrologicEngineeringCenter/HEC-FDA#119] [Q&A] [answered]
Is Model.Functions or Functions the most up to date code? Is there stuff in Model that is necessary or is it the old code that is partially refactored?

**Comment by HenryGeorgist** [comment]
@codymccoy @rnugent3 I could use some direction here, I think the entire Model project is more or less defunct - is that right?

**Reply by rnugent3** [reply]
I will take a look today. Your guess is probably as good if not better than mine.

**Reply by codymccoy** [reply]
I am not sure what you mean by "defunct". We are currently referencing the model from the VM. It will be defunct if you make a new model and swap it out.

**Reply by HenryGeorgist** [reply]
I saw that the model is included in the resources of the VM, and I found many classes that are never actually referenced. I struggled to find any classes that were specifically referenced in the VM. Many of the references point to things like Functions."star" or Statistics."star" but not Model."star"

**Reply by codymccoy** [reply]
I did a little poking around. The class in the model called "IFdaFunctionFactory" is used all over the view model.
![image](https://user-images.githubusercontent.com/47757848/136228480-bb75fbe3-737e-4caa-8f04-0592978321e8.png)

**Reply by HenryGeorgist** [reply]
@codymccoy I guess what I am concerned about is that I want to minimize your pain. I am trying to find the right classes to cover in the new model, and the right classes in the old model and related libraries to deprecate. I keep finding what seem to be code remnants so I am trying to reduce those to make sure I can efficiently support you.

**Reply by HenryGeorgist** [reply]
ah. That explains it. Everything is built through the factory. Perfect. Thanks!

**Comment by HenryGeorgist** [comment]
Based on @codymccoy 's answer i think we can take the new FDA model, implement the IFDAFunction interface on the UncertaintyPairedData and PairedData using an adapter, and then update the IFDAFunctionFactory to substitute the entire FDA model out in pretty much one fell swoop.
I would reccomend we then deprecate the IFDAFunctionFactory class, and fix the 27 references to point directly at the new FDA model and then delete the entire model project.  Thoughts? @rnugent3

**Reply by codymccoy** [reply]
Sounds good to me.

**Comment by rnugent3** [comment, marked as answer]
model has been deleted

**Reply by rnugent3** [reply]
and replaced with fda-model
